client/trino-cli/src/main/java/io/trino/cli/AlignedTablePrinter.java

```diff
@@ -94,12 +100,26 @@ public void printRows(List<List<?>> rows, boolean complete)
         int columns = fieldNames.size();
 
         int[] maxWidth = new int[columns];
+        int[] maxDecimalWidth = new int[columns];
         for (int i = 0; i < columns; i++) {
             maxWidth[i] = max(1, consoleWidth(fieldNames.get(i)));
         }
+
+        for (List<?> row : rows) {
+            for (int i = 0; i < row.size(); i++) {
+                String s = formatValue(row.get(i));
+                if (decimalFields.get(i) && row.get(i) != null) {
+                    maxDecimalWidth[i] = max(maxDecimalWidth[i], s.length() - s.indexOf('.'));
+                }
+            }
+        }
+
         for (List<?> row : rows) {
             for (int i = 0; i < row.size(); i++) {
                 String s = formatValue(row.get(i));
+                if (decimalFields.get(i) && row.get(i) != null) {
+                    s = s + repeat(" ", maxDecimalWidth[i] - (s.length() - s.indexOf('.')));
```

Comment:
```suggestion
                    maxDecimalWidth[i] = max(maxDecimalWidth[i], s.length() - s.indexOf('.'));
                    s = s + repeat(" ", maxDecimalWidth[i] - (s.length() - s.indexOf('.')));
```

This should allow to remove the previous loop.

Reply:
I'm not sure that will work because we need to add whitespace equal to the max decimalWidth in all decimals. Keeping them in the same loop will make it pad equal to the local maximum which might not work.

For example, consider the case of `[1.5, 1.543, 1.54]` will turn to `[1.5, 1.543, 1.54x]` instead of `[1.5xx, 1.543, 1.54x]`.